### PR TITLE
Add better documentation

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Improve the README.

- feat: update readme and license
- refactor: apply linter to itself
- fix: publish executable package
- docs: add more info about project

## How to test

Ask people what they think. Mainly @jamesgorrie?

## How can we measure success?

Adoption!

## Have we considered potential risks?

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
+# The Guardian Package Linter
 
-Validate your NPM dependencies without installing Node (WIP)
+Lint your `package.json` in line with our recommendations for
+[dependencies](https://github.com/guardian/recommendations/blob/main/dependencies.md#javascript)
+and
+[packages](https://github.com/guardian/recommendations/blob/main/npm-packages.md).
 
-[Publint](https://publint.dev/@guardian/package-linter)
+**Please note** that in its current implementation, this linter will overwrite
```
Comment (jamesgorrie) [inline README.md]: If you wanted fancy notes:
https://github.com/orgs/community/discussions/16925

File: README.md [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
+the `package.json` file that it processes.
 
-## How to get a health report for your `package.json`?
+Based on the value of the `private` field, the package will be interpreted as a
+`lib` or an `app`, and processed accordingly.
```
Comment (jamesgorrie) [inline README.md]: I get it but it's not explicit as to which is which, `private: app` & `public: lib`. There's no reference further in the documentation explaining the difference.

File: README.md [hunk 3]
```diff
@@ -1,46 +1,47 @@
...
-    https://deno.land/x/package_health/src/main.ts \
-    ./package.json  \
-    --cache
+## Usage
+
+### With Node
+
+```sh
+npm install @guardian/package-linter;
+npx package-linter ./package.json;
```
Comment (jamesgorrie) [inline README.md:23]: I didn't think you needed to run `npm install`, rather just `npx`?

File: README.md [hunk 4]
```diff
@@ -1,46 +1,47 @@
...
 ```
 
-> - The `--cache` flag should help with speedups
-> - The `--verbose` flag is very verbose
+### With Deno
+
+```sh
+deno run -A https://deno.land/x/guardian_package_linter@latest/src/cli.ts ./package.json
```
Comment (jamesgorrie) [inline README.md]: I think something might need to make 0.6.0 latest as running this 404s, running `deno run -A https://deno.land/x/guardian_package_linter@0.6.0/src/cli.ts ./package.json` doesn't.

File: README.md [hunk 5]
```diff
@@ -1,46 +1,47 @@
...
-- [x] Great error messaging
-- [ ] Investigate direct dependencies of dependencies, which may have peers
-      themselves
-- [ ] Give insight into possible resolution steps, including intersects
+This tool is still a work in progress, and here’s a list of things that we hope
+it can solve in the future
 
-## Shapes
+- [ ] Be explicit about missing peer dependencies and try installing them
+- [ ] Add a `--fix` flag and ensure it cannot be used in CI
```
Comment (jamesgorrie) [inline README.md:38]: Do you mean so that we can run this in CI without editing the file?

File: README.md [hunk 6]
```diff
@@ -1,46 +1,47 @@
...
+- [ ] Ensure that chosen licenses are appropriate
+- [ ] Improve distinctions between `app` and `lib`
```
Comment (jamesgorrie) [inline README.md]: My comment alludes to this, so maybe ignore.

File: fixtures/package.json [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
-  "dependencies": {
-    "@types/node": "~18.11",
-    "@guardian/source-foundations": "8.0.0",
-    "@guardian/core-web-vitals": "2.0.2"
-  },
-  "devDependencies": {
-    "@guardian/libs": "^12",
-    "@guardian/ab-react": "^2.0.1"
-  }
+	"name": "npm-dependencies",
```
Comment (jamesgorrie) [inline fixtures/package.json:2]:
```suggestion
	"name": "@guardian/package-linter/fixtures",
```

❓

**Review by jamesgorrie — APPROVED** [review]
Very nice!

File: README.md (hunk 1 above)
Comment (mxdvl) [reply README.md]: I worry about the accessibility implications: it’s not a quote!

File: README.md (hunk 3 above)
Comment (mxdvl) [reply README.md:23]: True, actually! I’m always confused about what `npx` actually does…

File: README.md (hunk 1 above)
Comment (jamesgorrie) [reply README.md]: I'm with you, I did think it was a weird choice.
It actually renders as `<p>` though.

Screenshot 2023-10-30 at 11 53 06

**Pull request merged**